chsdi/__init__.py

```diff
@@ -70,61 +74,277 @@ def main(global_config, **settings):
     initialize_sql(settings)
 
     # route definitions
-    config.add_route('dev', '/dev')
-    config.add_route('ga_api', '/loader.js')
-    config.add_route('testi18n', '/testi18n')
+    # Dev page with bunch of demo links
+    config.add_route('dev', '/dev', request_method='GET')
+    # JS API loader
+    config.add_route('ga_api', '/loader.js', request_method='GET')
+    add_cors_route(config, '/loader.js', 'ga_api', methods=['GET'])
+
+    config.add_route('testi18n', '/testi18n', request_method='GET')
+    # List topics
     config.add_route('topics', '/rest/services')
-    config.add_route('mapservice', '/rest/services/{map}/MapServer')
-    config.add_route('layersConfig', '/rest/services/{map}/MapServer/layersConfig')
-    config.add_route('catalog', '/rest/services/{map}/CatalogServer')
-    config.add_route('identify', '/rest/services/{map}/MapServer/identify')
-    config.add_route('find', '/rest/services/{map}/MapServer/find')
-    config.add_route('attribute_values', '/rest/services/{map}/MapServer/{layerId}/attributes/{attribute}')
-    config.add_route('legend', '/rest/services/{map}/MapServer/{layerId}/legend')
-    config.add_route('releases', '/rest/services/{map}/MapServer/{layerId}/releases')
-    config.add_route('cacheUpdate', '/rest/services/{map}/MapServer/{layerId}/cacheUpdate')
-    config.add_route('featureAttributes', '/rest/services/{map}/MapServer/{layerId}')
-    config.add_route('feature', '/rest/services/{map}/MapServer/{layerId}/{featureId}')
-    config.add_route('htmlPopup', '/rest/services/{map}/MapServer/{layerId}/{featureId}/htmlPopup')
-    config.add_route('iframeHtmlPopup', '/rest/services/{map}/MapServer/{layerId}/{featureId}/iframeHtmlPopup')
-    config.add_route('extendedHtmlPopup', '/rest/services/{map}/MapServer/{layerId}/{featureId}/extendedHtmlPopup')
-    config.add_route('search', '/rest/services/{map}/SearchServer')
-    config.add_route('wmtscapabilities', '/rest/services/{map}/1.0.0/WMTSCapabilities.xml')
-    config.add_route('feedback', '/feedback')
-    config.add_route('qrcodegenerator', '/qrcodegenerator')
-    config.add_route('sitemap', '/sitemap')
-    config.add_route('luftbilder', '/luftbilder/viewer.html')
-    config.add_route('historicalmaps', '/historicalmaps/viewer.html')
-    config.add_route('checker', '/checker')
-    config.add_route('checker_dev', '/checker_dev')
-    config.add_route('downloadkml', '/downloadkml')
-
-    config.add_route('files_collection', '/files', request_method=('GET', 'POST', 'DELETE'))
+    add_cors_route(config, '/rest/services', 'topics', methods=['GET'])
```

Comment:
Adding the CORS routes imho adds a lot of duplicated code, maybe it's worth packing the two into a generic method that adds normal and cors routes and makes it easier to see which routes are defined how?

Reply:
Hmm, https://github.com/geoadmin/mf-chsdi3/blob/d1ee3ff291d5b974ae6cb1ff7a8f0dbcd4240af8/chsdi/__init__.py#L25